### PR TITLE
Add reorg op

### PR DESCRIPTION
**Pull request by JiabinYang**

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -7455,6 +7456,58 @@ def maxout(x, groups, name=None):
...
+    Gives a stride to reorg the input tensor
+
+    Here are some example:
+
+    input is 4D LoDtensor with shape [batch, channel, height, width] and has an attrs stride = 2
+
+    reorg will do some math work to reorder the elements of input according to stride to construt
+    put with shape [batch, channel * stride * stride, height/stride, width/stride]
+
+    reorg is used to reorgnization the output of pre_layer and change the tensor to fit the shape
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py]: This sentence doesn't seem to contain information. Can you explain in more detail?

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -7455,6 +7456,58 @@ def maxout(x, groups, name=None):
...
+
+    Args:
+        x(variable): The input tensor.
+        stride(variable): The stride to reorg
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py]: What does "the stride to reorg" mean?

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -7455,6 +7456,58 @@ def maxout(x, groups, name=None):
...
 
+def reorg(x, stride, name=None):
+    """
+    Gives a stride to reorg the input tensor
+
+    Here are some example:
+
+    input is 4D LoDtensor with shape [batch, channel, height, width] and has an attrs stride = 2
+
+    reorg will do some math work to reorder the elements of input according to stride to construt
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py]: what's the "math"?

File: paddle/fluid/operators/reorg_op.cc [hunk 4]
```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qingqing01) [inline paddle/fluid/operators/reorg_op.cc]: 2016 ->  2018

**Review by qingqing01 — COMMENTED** [review]
LGTM.

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (JiabinYang) [reply python/paddle/fluid/layers/nn.py]: the formula is too complex to show and I add the more specific process of how this op work in latest commit

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/layers/nn.py]: done in latest commit

File: paddle/fluid/operators/reorg_op.cc (hunk 4 above)
Comment (JiabinYang) [reply paddle/fluid/operators/reorg_op.cc]: done in latest commit

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/fluid/layers/nn.py]: given more specific explanation

**Review by panyx0718 — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**